build_scripts/elk_setup/elk_setup.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Install Java
+yum -y install java-1.8.0-openjdk
+
+# Download and install Elasticsearch
+wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.2.rpm -P /tmp
+rpm -ivh /tmp/elasticsearch-5.6.2.rpm
+chkconfig --add elasticsearch
+
+# Configure Elasticsearch
+mv /etc/elasticsearch/elasticsearch.yml /etc/elasticsearch/elasticsearch.yml.orig
+ln -s /vagrant/build_scripts/elk_setup/elasticsearch.yml /etc/elasticsearch/elasticsearch.yml
+mkdir -p /data/es_logs
+chown -R vagrant:vagrant /data
+chmod 777 -R /data
+# Add the following lines to /etc/security/limits.conf for max number of threads error
+# elasticsearch - nproc 2048
+# elasticsearch - nofile 65536
+service elasticsearch start
```

Comment:
Don't we also need to do something so that ElasticSearch / Java uses IPv4 instead of IPv6?

Reply:
ElasticSearch will automatically use IPv4 when supply a proper IPv4 address in the `network_host` variable in the configuration file (elasticsearch.yml). Additionally, if `ES_JAVA_OPTS` is set to `-Djava.net.preferIPv4Stack=true` in the vagrant vm, an IPv6 socket will be used for IPv4. To truly have ElasticSearch use IPv4, IPv6 needs to be disabled at the OS level.

See https://github.com/elastic/elasticsearch/issues/24591.